File: backend/services/stockservice/src/main/java/com/api/stockservice/domain/Entities/Category.java
package com.api.stockservice.domain.Entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.GenericGenerator;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@DynamicUpdate
public class Category {
    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    private String Id;
    private String name;
    private String thumbnail;
}

File: backend/services/MarketplaceService/src/MarketplaceService.Domain/Entities/Category.cs
using System.ComponentModel.DataAnnotations;

namespace MarketplaceService.Domain.Entities
{
    public class Category
    {
        [Key]
        public string CategoryId { get; set; }

        public string Name { get; set; }

        public string Thumbnail { get; set; }
        public List<Product> Products { get; set; } = new List<Product>();
    }
}
